### PR TITLE
New weight meter PR

### PR DESCRIPTION
**Pull request by ermalkaleci**

### CODE REVIEW

**Review by xlc — APPROVED** [review]

**Review by shaunxw — COMMENTED** [review]
I think we need some unit tests like in `frame-support` to make sure all the macros expanded as expected.

Prints should be replaced with `log`.

File: weight-meter/src/lib.rs [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 use frame_support::weights::Weight;
 
+// For use in mock file
+extern crate self as orml_weight_meter;
```
Comment (xlc) [inline weight-meter/src/lib.rs:6]: interesting you can do this why is it needed here? how about `use crate as orml_weight_meter` in mock.rs?

**Review by xlc — APPROVED** [review]

File: weight-meter/src/lib.rs (hunk 1 above)
Comment (brettkolodny) [reply weight-meter/src/lib.rs:6]: That's what I had originally tried but for some reason I couldn't get it to work.

File: weight-meter/src/meter_std.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+thread_local! {
+	static METER: RefCell<Meter> = RefCell::new(Meter {
+		used_weight: 0,
+		depth: 0,
+	});
+}
+
+/// Start weight meter with base weight
+pub fn start_with() {
```
Comment (ermalkaleci) [inline weight-meter/src/meter_std.rs:15]:
```suggestion
pub fn start() {
```

**Pull request merged**